File: 0001~0500/0014. Longest Common Prefix/main.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int sz = strs.size()-1;
        string str = strs[0];
        int len = str.length();
        for (int i=1; i<=sz; i++) {
            int j;
            if (len > strs[i].length())
                len = strs[i].length();
            for (j=0; j<len; j++) {
                if (strs[i][j] == str[j]) 
                    continue;
                else
                    break;
            }
            len = j;
        }
        //cout<<len;
        string ans = str.substr(0,len);
        return ans;
    }
};

int main () {
    vector<string> strs;
    string str;
    while (cin>>str) {
        strs.push_back(str);
    }
    Solution s;
    cout<<s.longestCommonPrefix(strs);
}

File: 0001~0500/0014. Longest Common Prefix/main.py
"""
Source: https://leetcode.com/problems/longest-common-prefix/
Date: 2022/2/18
Constraints: 
    1 <= strs.length <= 200
    0 <= strs[i].length <= 200
    strs[i] consists of only lower-case English letters.
"""

from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ""
        i = 0
        while 1:
            if i < len(strs[0]):
                ch = strs[0][i]
            else:
                break
            valid = True
            for str in strs:
                if i >= len(str) or ch != str[i]:
                    valid = False
                    break
            if valid == True:
                ans += ch
                i += 1
            else:
                break
        return ans
                
        